src/sas/sascalc/dataloader/loader.py

```diff
@@ -59,98 +53,76 @@ def __init__(self):
-        except FileContentsException as e:
-            if debug: traceback.print_exc()
-            if msg_from_reader is None:
-                raise RuntimeError(e.message)
-        except Exception:
-            if debug: traceback.print_exc()
-            pass
-
-        # CanSAS XML reader failed - try NXcanSAS reader
-        try:
-            cansas_nexus_loader = cansas_reader_HDF5.Reader()
-            return cansas_nexus_loader.read(path)
-        except DefaultReaderException as e:
-            if debug: traceback.print_exc()
-            logging.error("No default loader can load the data")
-            # No known reader available. Give up and throw an error
-            if msg_from_reader is None:
+            # Use backup readers
+            try:
+                return self.load_using_generic_loaders(path)
+            except NoKnownLoaderException as nkgl_e:
+                if debug: traceback.print_exc()
+                logging.error(nkgl_e)
+                # No known reader available. Give up and throw an error
+                msg = str(nkgl_e)
+                msg += "\nUnknown data format: {}.\nThe file is not a ".format(
+                    path)
+                msg += "known format that can be loaded by SasView.\n"
+                raise NoKnownLoaderException(msg)
+            except FileContentsException as e:
+                if debug: traceback.print_exc()
+                raise RuntimeError(e)
+            except DefaultReaderException as e:
+                if debug: traceback.print_exc()
+                logging.error("No default loader can load the data")
+                # No known reader available. Give up and throw an error
                 msg = "\nUnknown data format: {}.\nThe file is not a ".format(path)
                 msg += "known format that can be loaded by SasView.\n"
                 raise NoKnownLoaderException(msg)
-            else:
-                # Associated reader and default readers all failed.
-                # Show error message from associated reader
-                raise RuntimeError(msg_from_reader)
-        except FileContentsException as e:
-            if debug: traceback.print_exc()
-            err_msg = msg_from_reader if msg_from_reader is not None else e.message
-            raise RuntimeError(err_msg)
+            except Exception as e:
+                if debug: traceback.print_exc()
+                raise RuntimeError(e)
+
+    def load_using_generic_loaders(self, path):
+        """
+        If the expected reader cannot load the file or no known loader exists,
+        attempt to load the file using a few defaults readers
+        :param path: file path
+        :return: List of Data1D and Data2D objects
+        """
+        module_list = readers.get_generic_readers()
+        for module in module_list:
+            reader = module.Reader()
+            try:
+                return reader.read(path)
```

Comment:
Maybe want to skip a "successful" read that didn't return any data as you do above?
```python
data_list = reader.read(path)
if data_list:
    return data_list
```